### PR TITLE
RefreshIndicator color, backgroundColor

### LINKED ISSUE #4922: RefreshIndicator should allow overriding backgroundColor
I want to color the background of the RefreshIndicator different than the backgroundColor of my theme. It would be great if RefreshIndicator supported this.

Thanks!

### PR DESCRIPTION
**Pull request by HansMuller**
Add support for specifying the foreground and background colors for a RefreshIndicator.
Fixes #4922

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/overscroll_demo.dart [hunk 1]
```diff
@@ -31,15 +31,18 @@ class OverscrollDemoState extends State<OverscrollDemo> {
...
-       content: new Text("Refresh complete"),
-       action: new SnackBarAction(
-         label: 'RETRY',
-         onPressed: () {
-           _refreshIndicatorKey.currentState.show();
-         }
-       )
-      ));
+      final ScaffoldState scaffoldState = _scaffoldKey.currentState;
+      if (scaffoldState != null) {
```
Comment (abarth) [inline examples/flutter_gallery/lib/demo/overscroll_demo.dart]: Why not `_scaffoldKey.currentState?.showSnackBar` ?

**Comment by abarth** [comment]
LGTM

File: examples/flutter_gallery/lib/demo/overscroll_demo.dart (hunk 1 above)
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/overscroll_demo.dart]: It's to handle the case where the user backs out of the page before the completer finishes.

Comment (abarth) [inline examples/flutter_gallery/lib/demo/overscroll_demo.dart]: I'm not sure I understand.  I just mean that the `?.` operator will do the null check for you.

Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/overscroll_demo.dart]: Oh. In that case, you're right, I should have used ?. :-).  Will fix.

**Pull request merged**